lib/json_api_client/resource.rb

```diff
@@ -431,6 +431,14 @@ def save
       end
     end
 
+    def message_for(error)
```

Comment:
Does this belong as an instance method? If it's extracted here, perhaps it belongs as a class method for this resource. It can still be overridden by a child class if you so choose.